platform/linux-generic/odp_packet_vector.c

```diff
@@ -55,3 +59,63 @@ void odp_packet_vector_free(odp_packet_vector_t pktv)
 
 	odp_buffer_free(odp_buffer_from_event(ev));
 }
+
+int odp_packet_vector_valid(odp_packet_vector_t pktv)
+{
+	odp_event_vector_hdr_t *pktv_hdr = _odp_packet_vector_hdr(pktv);
+	pool_t *pool = pktv_hdr->buf_hdr.pool_ptr;
+	uint32_t i;
+
+	if (odp_unlikely(pktv == ODP_PACKET_VECTOR_INVALID))
+		return 0;
+
+	if (odp_unlikely(pktv_hdr->size > pool->params.vector.max_size))
+		return 0;
+
+	for (i = 0; i < pktv_hdr->size; i++) {
+		if (pktv_hdr->packet[i] == ODP_PACKET_INVALID)
+			return 0;
+	}
+
+	return 1;
+}
+
+void odp_packet_vector_print(odp_packet_vector_t pktv)
+{
+	int max_len = 4096;
+	char str[max_len];
+	int len = 0;
+	int n = max_len - 1;
+	uint32_t i;
+	odp_event_vector_hdr_t *pktv_hdr = _odp_packet_vector_hdr(pktv);
+
+	len += snprintf(&str[len], n - len, "Packet Vector\n");
+	len += snprintf(&str[len], n - len,
+			"  handle %p\n", pktv);
+	len += snprintf(&str[len], n - len,
+			"  len    %" PRIu16 "\n", pktv_hdr->size);
```

Comment:
use API name for the size: len => size

size field in pktv_hdr is uint32_t => use PRIu32